File: delectable/src/main/java/com/delectable/restaurant/RestaurantService.java
package com.delectable.restaurant;

import com.delectable.shared.crud.CRUSoftDeleteRepository;
import com.delectable.shared.crud.CRUSoftDeleteService;

public class RestaurantService extends CRUSoftDeleteService<Restaurant>{
    public RestaurantService(CRUSoftDeleteRepository<Restaurant> repository) {
        super(repository);
    }
}

File: delectable-client/src/app/delectable/restaurant/service/restaurant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { CRUDService } from 'app/delectable/shared/service/crud.service';

const restaurantAPIEndpoint: string = "/api/restaurant";

@Injectable({
  providedIn: 'root'
})
export class RestaurantService extends CRUDService {

  constructor(http: HttpClient) {
    super(http, restaurantAPIEndpoint)
  }
  
}
